sourcegraph/sourcegraph
# feat(licensing): add "./api/license/check" dotcom api handler

Part of https://github.com/sourcegraph/sourcegraph/issues/52507.

This PR:
- Adds a new "./api/license/check" API handler to check whether the license (v2) is used only once per instance

## Test plan
- Check the diff
- Tests should pass

Manual testing
- `sg start dotcom`
- Connect to db and get any non-empty `license_check_token`: 
   - get hex_token `SELECT encode(license_check_token::bytea, 'hex') as hex_token  FROM product_licenses LIMIT 10`;
   - get site_id `SELECT site_id from global_state;`
- Call API endpoint

```bash
$ curl -i -H "Content-Type: application/json" \
    -H "Authorization: Bearer {hex_token}" \
    -X POST https://sourcegraph.test:3443/.api/license/check \
    -d '{"siteID":"site_id"}'

HTTP/2 200 
access-control-allow-credentials: true
access-control-allow-origin: 
cache-control: no-cache, max-age=0
content-type: application/json
date: Mon, 05 Jun 2023 16:55:46 GMT
server: Caddy
server: Caddy
set-cookie: sourcegraphDeviceId=5b0c3885-9daa-4807-b863-6d6877d9c5a9; Expires=Wed, 05 Jun 2024 16:55:46 GMT; Secure
vary: Cookie, Accept-Encoding, Authorization, Cookie, Authorization, X-Requested-With, Cookie
x-content-type-options: nosniff
x-frame-options: DENY
x-powered-by: Express
x-xss-protection: 1; mode=block
content-length: 54

{"error":null,"data":{"is_valid":true,"reason":null}}
```
